### PR TITLE
Lps 152963 4

### PR DESCRIPTION
**Pull request by SamZiemer**
https://issues.liferay.com/browse/LPS-152963

This pull addresses an issue where the role description column exceeds the maximum length allowed. I have upgraded the column to be a clob. In addition I have had to address some Oracle and DB2 query failures when using clobs. Please let me know if there are any questions. 

/cc @vicnate5

### CODE REVIEW

**Comment by vicnate5** [comment]
ci:test:environments

**Comment by melodywu** [comment]
No related failure found in ci:test:relevant.

**Comment by Sharryxue** [comment]
ci:test:upgrade-all

**Comment by Sharryxue** [comment]
No related failure found in ci:test:environments. Trigger the upgrade-all test suite since there is new upgrade process added, and changes in DB made.

**Comment by melodywu** [comment]
ci:test:upgrade-all

**Comment by vicnate5** [comment]
ci:test:upgrade-all:norebase

**Comment by vicnate5** [comment]
Upgrade-all results look good.

**Comment by drewbrokke** [comment]
@SamZiemer sorry about the delay on this one. I'll review today.

Also, looks like there's a merge conflict. Please resolve and force-push.

Thanks!

File: portal-impl/src/com/liferay/portal/upgrade/v7_4_x/PortalUpgradeProcessRegistryImpl.java [hunk 1]
```diff
@@ -228,6 +228,8 @@ protected String[][] getTableAndPrimaryKeyColumnNames() {
 			new Version(22, 0, 0),
 			UpgradeProcessFactory.alterColumnName(
 				"Organization_", "statusId", "statusListTypeId LONG"));
+
+		upgradeVersionTreeMap.put(new Version(22, 0, 1), new UpgradeRole());
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/upgrade/v7_4_x/PortalUpgradeProcessRegistryImpl.java]: @SamZiemer we can probably just inline this class using `UpgradeProcessFactory.alterColumnName`. Will you please check to see if that works?

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java [hunk 2]
```diff
@@ -1249,41 +1252,52 @@ public List<Role> getUserRelatedRoles(long userId, long[] groupIds) {
...
-						for (int i = 1; i < groupIds.length; i++) {
-							predicate = predicate.or(
-								Groups_RolesTable.INSTANCE.groupId.eq(
-									groupIds[i]));
-						}
-
-						return predicate.withParentheses();
+			));
+
+		if (!userRoles.isEmpty()) {
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:1257]: We should prefer `SetUtil.isNotEmpty()` for all the manual `list.isEmpty()` checks.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
Looks great, @SamZiemer, just the one comment about using `SetUtil` and the merge conflict. Thanks!

**Comment by SamZiemer** [comment]
hi @drewbrokke, I've made the changes requested. It looks like I may need to submit a new PR so I'll close this one and open a new one.

**Comment by SamZiemer** [comment]
New Pr https://github.com/liferay-usm/liferay-portal/compare/master...SamZiemer:lps-152963-4

File: portal-impl/src/com/liferay/portal/upgrade/v7_4_x/PortalUpgradeProcessRegistryImpl.java (hunk 1 above)
Comment (drewbrokke) [reply portal-impl/src/com/liferay/portal/upgrade/v7_4_x/PortalUpgradeProcessRegistryImpl.java]: Oops, sorry just saw you already did this in a later commit.

**Pull request closed without merging**